Composer/packages/client/src/recoilModel/dispatchers/formDialogs.ts

```diff
@@ -79,14 +79,37 @@ export const formDialogsDispatcher = () => {
       } catch (error) {
         set(applicationErrorState, {
           message: error.message,
-          summary: formatMessage('Generating form dialog using ${schemaId} schema failed.', { schemaId }),
+          summary: formatMessage('Generating form dialog using "{ schemaId }" schema failed.', { schemaId }),
         });
       } finally {
         set(formDialogGenerationProgressingState, false);
       }
     }
   );
 
+  const removeFormDialog = useRecoilCallback(
+    (callbackHelpers: CallbackInterface) => async ({ projectId, dialogId }) => {
+      const { set, snapshot } = callbackHelpers;
+
+      const dialog = await snapshot.getPromise(dialogState({ projectId, dialogId }));
+      const { reloadProject } = await snapshot.getPromise(dispatcherState);
+
+      try {
+        if (!dialog) {
+          return;
+        }
+
+        const response = await httpClient.delete(`/formDialogs/${projectId}/${dialogId}`);
+        await reloadProject(callbackHelpers, response);
```

Comment:
Do we need to do anything with the file state to prevent the delete from causing that issue with reloading (that thing you previously saw during generation)?